File: gen_asset_docs.py
import requests

url = "https://raw.githubusercontent.com/FuelLabs/verified-assets/main/ASSETS.md"

output_path = "./docs/src/assets.md"

try:
    response = requests.get(url)
    response.raise_for_status()

    content = "# Verified Assets\n\n" + response.text

    with open(output_path, 'w') as file:
        file.write(content)

    print(f"Content successfully written to {output_path}")

except requests.exceptions.RequestException as e:
    print(f"An error occurred while fetching the file: {e}")
except Exception as e:
    print(f"An error occurred: {e}")

File: src/gen-asset-docs.mjs
import fs from 'fs';
import fetch from 'node-fetch';

const url = "https://raw.githubusercontent.com/FuelLabs/verified-assets/main/ASSETS.md";
const outputPath = "./docs/src/assets.md";

// The custom section to be included
const customSection = `
# Verified Assets

## Using this section

You can find the current list of verified assets maintained by Fuel here: [verified-assets.json](https://verified-assets.fuel.network/assets.json)

Projects are welcome to use this information, but please note that it is provided at your own risk.

Additionally, you can download the latest asset information and icons in a single archive. This is useful if you want to locally cache the list or include it in a release pipeline for your tools and libraries: [verified-assets.zip](https://github.com/FuelLabs/verified-assets/)

For more information, please visit the verified assets repository [here](https://github.com/FuelLabs/verified-assets/).
`;

// Main function to fetch and write content
const fetchAndWriteContent = async () => {
    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP error! Status: ${response.status}`);
        }

        const text = await response.text();
        // Prepend the custom section to the fetched content
        const content = customSection + '\n' + text;

        // Write the content to the specified file
        fs.writeFileSync(outputPath, content);
        console.log(`Content successfully written to ${outputPath}`);
        
    } catch (error) {
        console.error(`An error occurred: ${error}`);
    }
};

// Run the main function
fetchAndWriteContent();
